### DISCUSSION TITLE
Is it possible to predict multiple timesteps?

### DISCUSSION THREAD

**Discussion by AlwaysRedNeverGreen** [neuraloperator/neuraloperator#369] [Q&A]
I'm currently using the library to make time step based predictions.  Does the library have the capability to predict multiple time steps into the future? (T+2,T+3...) or am i just missing something

**Comment by dhpitt** [comment]
Multiple timesteps into the future as in, given a(x,t), predict u(x,t+2)? Or as in given a(x,t) predict u(x,t) and u(x,t+1) at the same time?

**Comment by AlwaysRedNeverGreen** [comment]
Given an input T, predict T+1, then use it as the input to predict T+2 and so on. Hopefully that makes sense

**Comment by dhpitt** [comment]
The library doesn't currently have a training module written for the autoregressive use case

**Comment by gokhalen** [comment]
https://github.com/neuraloperator/neuraloperator/blob/master/fourier_2d_time.py  

This might be what you're looking for.

**Reply by Penido-exe** [reply]
The link is now broken. Can you help me finding it?

**Comment by Penido-exe** [comment]
Is it possible?

**Reply by JeanKossaifi** [reply]
Yes. You can either do auto-regressive prediction and feed your model's prediction back to it as input, or you can use a 3D FNO to predict a spatio-temporal volume.
